src/shared/loop-util.c

```diff
@@ -572,6 +627,15 @@ static int loop_device_make_internal(
                 (void) usleep_safe(usec);
         }
 
+        discard_max_bytes = fd_get_max_discard(fd);
+        /* If we could not determine discard_max_bytes, e.g. if the fd is a regular file not a block device
+         * or when running in a chroot without /sys mounted, keep loop dev defaults */
```

Comment:
we usually prefer these days if there's debug logging of errors we knowingly ignore. i.e.:

```c
r = fd_get_max_discard(fd, &discard_max_bytes);
if (r < 0)
        log_debug_errno(r, "Failed to read 'discard_max_bytes' of backing device, ignoring: %m");
else {
        …
}
``